EOF on ClientClickWindowPacket

I somehow got this error when clicking items on my chest inventory menus.
Still don't know why this happened.
The old buffer works without this error, weird.
```
ERROR [2022-10-29 05:25:13] - Throwable: java.io.EOFException: java.lang.RuntimeException: java.io.EOFException
        at net.minestom.server.network.NetworkBufferTypes.lambda$static$25(NetworkBufferTypes.java:253)
        at net.minestom.server.network.NetworkBuffer.read(NetworkBuffer.java:101)
        at net.minestom.server.network.NetworkBufferTypes.lambda$static$33(NetworkBufferTypes.java:316)
        at net.minestom.server.network.NetworkBuffer.read(NetworkBuffer.java:101)
        at net.minestom.server.network.packet.client.play.ClientClickWindowPacket.<init>(ClientClickWindowPacket.java:23)
        at net.minestom.server.network.packet.client.ClientPacketsHandler.create(ClientPacketsHandler.java:35)
        at net.minestom.server.network.PacketProcessor.create(PacketProcessor.java:32)
        at net.minestom.server.network.PacketProcessor.process(PacketProcessor.java:45)
        at net.minestom.server.network.player.PlayerSocketConnection.lambda$processPackets$0(PlayerSocketConnection.java:109)
        at net.minestom.server.utils.PacketUtils.readPackets(PacketUtils.java:230)
        at net.minestom.server.network.player.PlayerSocketConnection.processPackets(PlayerSocketConnection.java:103)
        at net.minestom.server.network.socket.Worker.lambda$run$0(Worker.java:79)
        at java.base/sun.nio.ch.SelectorImpl.processReadyEvents(SelectorImpl.java:294)
        at java.base/sun.nio.ch.EPollSelectorImpl.processEvents(EPollSelectorImpl.java:195)
        at java.base/sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:135)
        at java.base/sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:129)
        at java.base/sun.nio.ch.SelectorImpl.select(SelectorImpl.java:161)
        at net.minestom.server.network.socket.Worker.run(Worker.java:59)
    Caused by: java.io.EOFException
        at java.base/java.io.DataInputStream.readUnsignedShort(DataInputStream.java:346)
        at java.base/java.io.DataInputStream.readUTF(DataInputStream.java:595)
        at java.base/java.io.DataInputStream.readUTF(DataInputStream.java:570)
        at org.jglrxavpok.hephaistos.nbt.NBTString$Companion.readContents(NBTString.kt:39)
        at org.jglrxavpok.hephaistos.nbt.NBTString$Companion.readContents(NBTString.kt:37)
        at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readTag(Extensions.kt:27)
        at org.jglrxavpok.hephaistos.nbt.NBTList$Companion.readContents(NBTList.kt:88)
        at org.jglrxavpok.hephaistos.nbt.NBTList$Companion.readContents(NBTList.kt:77)
        at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readTag(Extensions.kt:27)
        at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readFullyFormedTag(Extensions.kt:18)
        at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion$readContents$$inlined$Kompound$1.run(NBT.kt:224)
        at org.jglrxavpok.hephaistos.nbt.NBT$Companion.Compound(NBT.kt:110)
        at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion.readContents(NBTCompound.kt:98)
        at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion.readContents(NBTCompound.kt:60)
        at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readTag(Extensions.kt:27)
        at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readFullyFormedTag(Extensions.kt:18)
        at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion$readContents$$inlined$Kompound$1.run(NBT.kt:224)
        at org.jglrxavpok.hephaistos.nbt.NBT$Companion.Compound(NBT.kt:110)
        at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion.readContents(NBTCompound.kt:98)
        at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion.readContents(NBTCompound.kt:60)
        at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readTag(Extensions.kt:27)
        at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readFullyFormedTag(Extensions.kt:18)
        at org.jglrxavpok.hephaistos.nbt.NBTReader.readNamed(NBTReader.kt:48)
        at org.jglrxavpok.hephaistos.nbt.NBTReader.read(NBTReader.kt:58)
        at net.minestom.server.network.NetworkBufferTypes.lambda$static$25(NetworkBufferTypes.java:251)
        ... 17 more
```

--- a/src/main/java/net/minestom/server/network/NetworkBufferTypes.java
+++ b/src/main/java/net/minestom/server/network/NetworkBufferTypes.java
@@ -242,7 +242,12 @@ final class NetworkBufferTypes {
                     nbtReader = new NBTReader(new InputStream() {
                         @Override
                         public int read() {
-                            return buffer.read(BYTE);
+                            return buffer.read(BYTE) & 0xFF;
+                        }
+
+                        @Override
+                        public int available() {
+                            return buffer.readableBytes();
                         }
                     }, CompressedProcesser.NONE);
                     buffer.nbtReader = nbtReader;

